Doc/library/asyncio-task.rst

```diff
@@ -602,6 +602,86 @@ Waiting Primitives
            # ...
 
 
+Running in Threads
+==================
+
+.. coroutinefunction:: to_thread(func, /, \*args, \*\*kwargs)
+
+   Asynchronously run function *func* in a separate thread.
+
+   Any \*args and \*\*kwargs supplied for this function are directly passed
+   to *func*.
+
+   Return an :class:`asyncio.Future` which represents the eventual result of
+   *func*.
+
+   This coroutine function is primarily intended to be used for executing
+   IO-bound functions/methods that would otherwise block the event loop if
+   they were ran in the main thread. For example, the following code would
+   block the event loop::
+
+       # "async def" is just used here so we can submit to asyncio.gather()
+       async def blocking_io():
```

Comment:
The documentation is often used as a source for copy-paste without careful reading.
I think adding a wrong code snippet without a very special markup (like red never-do-it text label) can be confusing.

I suggest keeping the only correct code example. Adding `.. warning::` section with a caution about blocking code inside a coroutine might be a good idea also.

Reply:
> The documentation is often used as a source for copy-paste without careful reading.

Great point, I had not honestly not considered that users could potentially copy-paste that example without reading the context.

However, I do think that there's some significant value in the time comparisons. I largely credit the `create_task()` examples w/ time comparisons at the start of the page for greatly improving my own understanding of how tasks work. Especially with something like async, it's not always easy to grasp without some visualization.

I think what I'll do is remove the first example, and leave an after remark about the current second example saying something like:

"Directly calling `blocking_io()` would block the event loop for its duration, resulting in an additional 1 second of run time. Instead, by using `asyncio.to_thread()`, we can run it in a separate thread without blocking the event loop."

(I'll also of course revise the paragraph before the example.)

Thanks for the feedback.

Reply:
I don't know how soon the release will be cut, but you might want to consider merging now, and honing the docs in a follow-up PR. Otherwise, there's a chance you can miss the deadline.

Reply:
> Adding .. warning:: section with a caution about blocking code inside a coroutine might be a good idea also.

That is certainly good asyncio advice, but I'm not certain that it would fit well with the rest of the current `asyncio.to_thread()` documentation. Perhaps we could consider placing something like that somewhere else, and linking to `asyncio.to_thread()`? 

To me, it seems like it would be more appropriately placed within a more general section that explains proper coroutine usage in asyncio.

Reply:
I'll go ahead and apply automerge now, good point @cjerdonek. I think the docs are now adequate after the last commit, we can improve further at a later time if needed.

Reply:
Agree